### ISSUE TITLE
smg.plot_corr(corr_matrix, xnames=corr_columns.columns) is raising ValueError

### ISSUE THREAD

**Issue by JanSchulz** [statsmodels/statsmodels#2510]
Using columns as `xnames` argument to `plot_corr` raises:

```
df_business = pandas.DataFrame(...)
corr_columns = df_business[<list_of_columns>]
corr_matrix = corr_columns.corr()
smg.plot_corr(corr_matrix, xnames=corr_columns.columns)
plt.show()
```

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-24-f81f43c530b6> in <module>()
      1 corr_columns = df_business[...]
      2 corr_matrix = corr_columns.corr()
----> 3 smg.plot_corr(corr_matrix, xnames=corr_columns.columns)
      4 plt.show()

C:\portabel\miniconda\envs\sc\lib\site-packages\statsmodels\graphics\correlation.pyc in plot_corr(dcorr, xnames, ynames, title, normcolor, ax, cmap)
     89     labelPos = np.arange(0, nvars) + 0.5
     90 
---> 91     if ynames:
     92         ax.set_yticks(labelPos)
     93         ax.set_yticks(labelPos[:-1]+0.5, minor=True)

C:\portabel\miniconda\envs\sc\lib\site-packages\pandas\core\index.pyc in __nonzero__(self)
   1039         raise ValueError("The truth value of a {0} is ambiguous. "
   1040                          "Use a.empty, a.bool(), a.item(), a.any() or a.all()."
-> 1041                          .format(self.__class__.__name__))
   1042 
   1043     __bool__ = __nonzero__

ValueError: The truth value of a Index is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

**Comment by josef-pkt** [comment]
This looks like a prio-high bug for 0.7, but I don't have time right now to check.

My guess is that there should be an `is None` check.

(PR welcome, I will be back online in several hours.)

**Comment by josef-pkt** [comment]
this might currently assume that xname is a python list and doesn't have checks for any other type of iterable.

**Comment by JanSchulz** [comment]
merged in #2512
